src/app/loadout/LoadoutDrawer.tsx

```diff
@@ -520,44 +580,63 @@ function LoadoutDrawer({
-            onDroppedItem={onAddItem}
-          >
-            {warnitems.length > 0 && (
-              <div className="loadout-contents">
-                <p>{t('Loadouts.VendorsCannotEquip')}</p>
-                <div className="loadout-warn-items">
-                  {warnitems.map((item) => (
-                    <div key={item.id} className="loadout-item">
-                      <InventoryItem item={item} onClick={() => fixWarnItem(item)} />
-                      <div className="close" onClick={() => onRemoveItem(item)} />
-                    </div>
-                  ))}
+    <>
+      <Sheet onClose={close} header={header}>
+        <div id="loadout-drawer" className="loadout-create">
+          <div className="loadout-content">
+            <LoadoutDrawerDropTarget
+              bucketTypes={bucketTypes}
+              storeIds={stores.map((s) => s.id)}
+              onDroppedItem={onAddItem}
+            >
+              {warnitems.length > 0 && (
+                <div className="loadout-contents">
+                  <p>{t('Loadouts.VendorsCannotEquip')}</p>
+                  <div className="loadout-warn-items">
+                    {warnitems.map((item) => (
+                      <div key={item.id} className="loadout-item">
+                        <InventoryItem item={item} onClick={() => fixWarnItem(item)} />
+                        <div className="close" onClick={() => onRemoveItem(item)} />
+                      </div>
+                    ))}
+                  </div>
                 </div>
+              )}
+              <div className="loadout-contents">
+                <LoadoutDrawerContents
+                  loadout={loadout}
+                  savedMods={savedMods}
+                  items={items}
+                  defs={defs}
+                  buckets={buckets}
+                  stores={stores}
+                  itemSortOrder={itemSortOrder}
+                  equip={onEquipItem}
+                  remove={onRemoveItem}
+                  add={onAddItem}
+                  onOpenModPicker={(query?: string) =>
+                    stateDispatch({ type: 'openModPicker', query })
+                  }
+                  removeModByHash={removeModByHash}
+                />
               </div>
-            )}
-            <div className="loadout-contents">
-              <LoadoutDrawerContents
-                loadout={loadout}
-                items={items}
-                defs={defs}
-                buckets={buckets}
-                stores={stores}
-                itemSortOrder={itemSortOrder}
-                equip={onEquipItem}
-                remove={onRemoveItem}
-                add={onAddItem}
-              />
-            </div>
-          </LoadoutDrawerDropTarget>
+            </LoadoutDrawerDropTarget>
+          </div>
         </div>
-      </div>
-    </Sheet>
+      </Sheet>
+      {modPicker.show &&
+        defs.isDestiny2() &&
+        ReactDOM.createPortal(
+          <ModPicker
```

Comment:
Technically this could be anywhere because of the portal, it doesn't *need* to be a sibling of the loadout sheet.

Reply:
Done